### DISCUSSION TITLE
Description of baseline code and some ideas

### DISCUSSION THREAD

**Discussion by yukiregista** [yukiregista/DSC_STUDY#41] [Ideas]
The baseline that we have at FinDataChallenge/takazawa/notebooks/baseline_target_encoding_cv.ipynb has the following properties:
- Numerical features or categorical features with low number of categories are input as they are
- Some categorical features (`Sector`, `State`, `BankState`) are target encoded (be careful about this when training...! It should be combined with appropriate cv (or train/valid splitting) strategy... (see notebook)
- `Franchise` are coarsened into 0, 1, and others
- LightGBM is used for inference.
- It scores around **0.685** in the leaderboard (although I did not really fix seed so you might need some try that scores better/worse).

There are many places we can improve. Some possiblities are:
- Hyperparameter tuning with optuna
  - With optuna, they do NOT tune for `pos_scale` (class_weight). Tuning for this parameter might give much better result since our data is very imbalanced.
- Inclusion of oversampling techniques (as Kondo-sensei's found out, SMOTEENN seems to be good for this task)
- Addition of some engineered features. Possibilities include:
  - Some features created by Chen-san (SameState, GrAppv/SBAApprv, Difference of date columns etc)
  - Alternative data
  - Features created by multiplication / division / or other multivariate functions 
- Inclusion of more encodings including:
  - Target encoding for more variables
  - Count/ Count_rank encoding
  - ...

**Please add replies below if you have suggestions!**
